app/vlinsert/opentelemetry/opentelemetry.go

```diff
@@ -138,3 +133,22 @@ func pushFieldsFromScopeLogs(sc *pb.ScopeLogs, commonFields []logstorage.Field,
 	}
 	return fields
 }
+
+func appendAttributes(fields []logstorage.Field, attributes []*pb.KeyValue, parentField string) []logstorage.Field {
+	for _, attr := range attributes {
+		fieldName := attr.Key
+		if parentField != "" {
+			fieldName = parentField + "." + fieldName
```

Comment:
At this point, memory is allocated for each nested field. We could add buffers here to reuse memory, but that would complicate the code. Let me know if you think we should add the buffers.